plugins/CoreAdminHome/Commands/PurgeOldArchiveData.php

```diff
@@ -134,9 +136,24 @@ private function getDatesToPurgeFor(InputInterface $input)
                 $dates[] = Date::factory($year . '-' . $month . '-' . '01');
             }
         } else {
+            $includeYearArchives = $input->getOption('include-year-archives');
+
             foreach ($dateSpecifier as $date) {
-                $dates[] = Date::factory($date);
+                $dateObj = Date::factory($date);
```

Comment:
(as comment above, it seems this logic should be in the class that invalidates and done in scheduled task as well instead of just in command)